File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/game/GameViewModel.java
package fr.erban.dxitcompanion.db.game;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;

import fr.erban.dxitcompanion.game.GameBean;

public class GameViewModel extends AndroidViewModel {

    private final GameRepository gameRepository;

    public GameViewModel(Application application) {
        super(application);
        gameRepository = new GameRepository(application);
    }

    public void insert(GameBean game) {
        gameRepository.insert(game);
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/game/GameViewModel.kt
package fr.erban.dxitcompanion.db.game

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import fr.erban.dxitcompanion.db.DxitDatabase
import fr.erban.dxitcompanion.game.GameBean
import kotlinx.coroutines.launch

class GameViewModel(application: Application) : AndroidViewModel(application) {

    private val gameRepository = GameRepository(application)
    val finishedGames: LiveData<List<GameEntity>> =
        DxitDatabase.getDatabase(application).gameDao().getAllFinished()

    fun insert(game: GameBean) {
        viewModelScope.launch {
            gameRepository.insert(game)
        }
    }
}
